cli.go

```diff
@@ -742,6 +742,145 @@ func main() {
+						}
+
+						Debugf("Search %s has returned %v repos", ShakespeareBG(query), len(repos))
+					RepoLoop:
+						for _, repo := range repos {
+							//repoURLs = append(repoURLs, repo.GetFullName()) // e.g. "kubernetes/dashboard"
+							isFork := repo.GetFork()
+							// "Currently we do not support analysis of forks. Consider adding the parent of the fork instead."
+							if isFork {
+								Warnf("Skipping fork %s", repo.GetFullName())
+								continue RepoLoop
+							}
+
+							repoURLs = append(repoURLs, repo.GetHTMLURL())    // e.g. "https://github.com/kubernetes/dashboard"
+							repoNames = append(repoNames, repo.GetFullName()) // e.g. "kubernetes/dashboard"
+						}
+					}
+
+					toBeFollowed := make([]string, 0)
+					if !noCheck { // exclude already-followed projects:
+						took := NewTimer()
+						Infof("Getting list of followed projects...")
+						projects, protoProjects, err := client.ListFollowedProjects()
+						if err != nil {
+							panic(err)
+						}
+						Infof("Currently %v projects (and %v proto) are followed; took %s", len(projects), len(protoProjects), took())
+
+						for _, repoURL := range repoURLs {
+							_, isFollowed := isAlreadyFollowedProject(projects, repoURL)
+							_, isFollowedProto := isAlreadyFollowedProto(protoProjects, repoURL)
+							isNOTFollowed := !isFollowed && !isFollowedProto
+							if isNOTFollowed {
+								toBeFollowed = append(toBeFollowed, repoURL)
+							}
+						}
+						toBeFollowed = Deduplicate(toBeFollowed)
+					} else {
+						toBeFollowed = Deduplicate(repoURLs)
+						repoNames = Deduplicate(repoNames)
+					}
+
+					totalToBeFollowed := len(toBeFollowed)
+					Infof("Will follow %v projects...", totalToBeFollowed)
+					if !force {
+						CLIMustConfirmYes("Do you want to continue?")
+					}
+
+					// Write toBeFollowed to temp file:
+					saveTargetListToTempFile("follow-by-code-search", toBeFollowed)
+
+					followedNew := 0
+
+					etac := eta.New(int64(totalToBeFollowed))
+
+					// follow repos:
+					for _, repoURL := range toBeFollowed {
+						envelope := follower(repoURL, etac)
+						if envelope != nil {
+							// if the project was NOT already known to lgtm.com,
+							// sleep to avoid triggering too many new builds:
+							isNew := !envelope.IsKnown()
+							if isNew {
+								followedNew++
+								// sleep:
+								time.Sleep(waitDuration)
+							}
+						}
+					}
+					Successf("Followed %v projects (%v new)", totalToBeFollowed, followedNew)
+
+					// add them to list
+					totalRepoNames := len(repoNames)
+					for len(repoNames) != 0 {
+						for repoNameIndex, repoName := range repoNames {
+							for _, projectListKey := range projectListKeys {
+								success, err := client.AddProjectToSelectionFromRepoName(projectListKey, repoName)
+								if err != nil {
+									panic(err)
+								}
+								if !success {
+									Warnf("Retrying later %s...", repoName) // Repo is new and hasn't been analysed yet.
```

Comment:
While loop to wait until every project has been added successfully. Perhaps adding an sleep each time an iteration is finished could reduce some load on lgtm.

Reply:
Used a downward loop that works fine :)